src/deploy-commands.mts

```diff
@@ -0,0 +1,58 @@
+// Deploy commands used by the bots. This is a separate script that must be run
+// everytime we update commands.
+// Note: If someone runs this on a separate fork targeting the
+// same server, please be aware it will completely override any previous changes
+// WARNING: this is heavily rate limited and this should be run infrequently
+import { REST } from "discord.js";
+import { Routes } from "discord-api-types/v9";
+import * as dotenv from "dotenv";
+import * as fs from "fs";
+import * as path from "path";
+import { Command } from "./types.mjs";
+import { dirname } from "path";
+import { fileURLToPath, pathToFileURL } from "url";
+dotenv.config(); // Load env parameters
+const __dirname = dirname(fileURLToPath(import.meta.url));
+
+const commands = [];
+// Grab all the command files from the commands directory you created earlier
+const foldersPath = path.join(__dirname, "commands");
+const commandFolders = fs.readdirSync(foldersPath);
+
+for (const folder of commandFolders) {
+    // Grab all the command files from the commands directory you created earlier
+    const commandsPath = path.join(foldersPath, folder);
+    const commandFiles = fs.readdirSync(commandsPath).filter(file => file.endsWith(".mjs")); // Compiled to js!
```

Comment:
I'm not eager about reading from the filesystem to determine which commands are available. I'd prefer to have an array with a list of paths to each command. Intelisense also greatly benefits this approach, as it'll be able to detect the export type of each command, to verify if it's valid to use in the `commands.push(command.data.toJSON());` line.

Reply:
Really? I'd rather have it automatically cipher through each file as it would begin to balloon up that const array if we started to add more